### PR TITLE
DS-189 Table Caption

### PR DESCRIPTION
**Pull request by cjwhitedev**
## Jira

[https://pegadigitalit.atlassian.net/browse/DS-189](https://pegadigitalit.atlassian.net/browse/DS-189)

## Summary

Add support for table captions.

## Details

- Add `caption` Twig prop.
- Fix JS error previously thrown if table contained a `<caption>`.
- Fully support `<caption>` by wrapping table in `<figure>` and caption in `<figcaption>` when `caption` is present.

> Note: caption is moved out of the main table content into `<figcaption>` in order to make it span the table container width, not the table width, which can overflow the container.

## How to test

- Review code changes
- Review [Table Caption demo page](https://feature-ds-189-table-caption.boltdesignsystem.com/pattern-lab/?p=components-table-caption)

### CODE REVIEW

**Comment by danielamorse** [comment]
@mikemai2awesome addressed the feedback about the images and WC demos:
> For the docs, you changed fillmurray to placeholder, can you change to a bolt-image instead? The point is to show different types of content. And for the WC docs, use web component markup, no twig please.

**Review by mikemai2awesome — CHANGES_REQUESTED** [review]

**Review by mikemai2awesome — APPROVED** [review]

**Comment by mikemai2awesome** [comment]
@cjwhitedev @danielamorse I fixed some indents and demo markup. Make sure your text editor is set to use tab width of 2 spaces.

File: packages/components/bolt-table/src/table.js [hunk 1]
```diff
@@ -143,12 +157,20 @@ class BoltTable extends BoltElement {
...
+    const captionClasses = cx('c-bolt-table__caption');
     let boltTableMarkup = [];
 
     const boltTable = this.convertElements(parseCode);
 
     function setSectionTag(tag) {
       switch (tag) {
+        case 'caption':
+          return html`
+            <!-- <caption class=${captionClasses}>
```
Comment (colbytcook) [inline packages/components/bolt-table/src/table.js:170]: Should we remove these commented tags?

File: packages/components/bolt-table/src/table.scss [hunk 2]
```diff
@@ -54,14 +54,14 @@ $bolt-table-row-hover-background-color: bolt-theme(text, 0.025);
...
-    caption-side: bottom;
-  }
+  // caption {
+  //   @include bolt-font-size(small);
+  //   color: bolt-theme(text, 0.65);
+  //   text-align: center;
+  //   background-color: bolt-theme(background);
+
+  //   caption-side: bottom;
+  // }
```
Comment (colbytcook) [inline packages/components/bolt-table/src/table.scss:64]: Should we remove this commented out code?

**Review by colbytcook — APPROVED** [review]

**Pull request merged**